code_samples/logging.js

```diff
@@ -0,0 +1,79 @@
+var winston = require('winston');
+var Config = require('../.').Config;
+var HazelcastClient = require('../.').Client;
+
+if(process.argv.length != 3){
+    console.log('You must give exactly one argument to run this tool!!!');
+}else{
+    switch (process.argv[2]){
+       
+        case 'winstonLogging.js':
+            var winstonAdapter = {
+                logger: new (winston.Logger)({
+                    transports: [
+                        new (winston.transports.Console)()
+                    ]
+                }),
+
+                levels: [
+                    'error',
+                    'warn',
+                    'info',
+                    'debug',
+                    'silly'
+                ],
+
+                log: function(level, className, message, furtherInfo) {
+                    this.logger.log(this.levels[level], className + ' ' + message);
+                }
+            };
+
+            var loggingHappened = false;
+
+
+            winstonAdapter.logger.on('logging', function(transport, level, msg, meta) {
+                loggingHappened = true;
+            });
+
+            //winston logging is active in case of winston object given as property
+            var cfg = new Config.ClientConfig();
+            cfg.properties['hazelcast.logging'] = winstonAdapter;
+
+
+            HazelcastClient.newHazelcastClient(cfg).then(function(client){
+                console.log('Log via winston logger: '+ loggingHappened);
+                client.shutdown();
+            })
+            break;
+       
+        case 'defaultLogging.js':
+            var cfg = new Config.ClientConfig();
+            cfg.properties['hazelcast.logging'] = 'default';
+
+
+            HazelcastClient.newHazelcastClient(cfg).then(function(client){
+                console.log('logging by default');
+                client.shutdown();
+            });
+            break;
+       
+        case 'emptyLogging.js':
+            HazelcastClient.newHazelcastClient().then(function(client){
+                console.log('logging by default when no configuration is given');
+                client.shutdown();
+            })
+            break;
+       
+        case 'offLogging.js':
+            HazelcastClient.newHazelcastClient().then(function(client){
+                console.log('There is no logging when configuration is off');
+                client.shutdown();
+            })
+            break;
+        
+        default:
+            console.log('Options for logging are '+'1-)winstonLogging.js, 2-)defaultLogging.js, 3-)emptyLogging.js, 4-)offLogging.js' +
```

Comment:
I think arguments does not need to end with `Logging.js`. `winston`, `default`, `empty`, and `off` are okay.